src/taskgraph/config.py

```diff
@@ -157,6 +157,9 @@ def validate_graph_config(config):
 
 
 def load_graph_config(root_dir):
+    # ensure we have an absolute path; this is required for assumptions
+    # made later, such as the `vcs_root` being a directory above `root_dir`
+    root_dir = os.path.abspath(root_dir)
```

Comment:
In theory, I think this should go in the `__post_init__` method of `GraphConfig`. In practice, I don't think it really matters. Given GraphConfig is frozen, it would mean we'd need to use `object.__setattr__` to set it, so this is probably fine.

Reply:
ah; but if there's ever anything else that creates a `GraphConfig` that could be a problem, I think? This seems like an easy change to make.

Reply:
Thanks!
https://stackoverflow.com/a/73718478